### ISSUE TITLE
`softmax`函数出现NaN

### ISSUE THREAD

**Issue by sijunhe** [PaddlePaddle/Paddle#55685]
### bug描述 Describe the Bug

`softmax`函数会出现nan的问题

```
import paddle
import paddle.nn.functional as F

rand_logits_int = paddle.randint(0, 30, (1, 100000)).astype("float32")
rand_logits_small_numbers = rand_logits_int + -330.0
F.softmax(rand_logits_int) # valid answer
F.softmax(rand_logits_small_numbers)  # all NaN
F.softmax(rand_logits_small_numbers - rand_logits_small_numbers.max()) # valid answer
```

### 其他补充信息 Additional Supplementary Information

```
>>> paddle.version.show()
full_version: 2.5.0
major: 2
minor: 5
patch: 0
rc: 0
cuda: 11.2
cudnn: 8.2.1
xpu: False
xpu_xccl: False
cinn: False
```
